source/gui/startupDialogs.py

```diff
@@ -192,8 +192,10 @@ def onContinueRunning(self, evt):
 		self.Destroy()
 		core.doStartupDialogs()
 
-	def onExit(self, evt):
-		core.triggerNVDAExit()
+	def onExit(self, evt: wx.CommandEvent):
+		if not core.triggerNVDAExit():
+			log.error("NVDA already in process of exiting, this indicates a logic error.")
+		self.Destroy()  # Without this, the onExit is called multiple times by wx.
```

Comment:
`triggerNVDAExit` only queues the `doShutdown` function. Many of the other callbacks call `Destroy` directly, so if doing that is a problem, there are plenty here. 
An alternative would be to `Unbind` the event handler within `onExit` so it can't be called a second time.

Reply:
`onExit` was being called multiple times, adding `wx.Destroy` resolves this, likely because all event bindings are deleted.

#### First time cause
When the cancel button is pressed.

#### Second time
I suspect due to the cancel button using `wx.ID_CANCEL`. I think the `wx.Dialog` uses this as a default action when the dialog is closed (eg via close button).
Sequence:
- `triggerNVDAExit` queues `_doShutdown`.
- Later, when `_doShutdown` is run, it calls `_closeAllWindows`
- `_closeAllWindows` uses `wx.CallAfter` to close top level windows
- During `wx.Yield` the windows are closed, calling the `onExit` event handler for the `LauncherDialog`